src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import * as api from '../../api'
+
+type Props = {
+  selectedTodo: Todo | null,
+  handleCloseModal: () => void,
+}
+
+export const TodoModal: React.FC<Props> = React.memo(({ selectedTodo, handleCloseModal }) => {
+
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (!selectedTodo) {
+      setUser(null);
+      return;
+    }
+
+    api.getUser(selectedTodo.userId).then(setUser);
```

Comment:
The loading state is not updated after fetching user data. Consider adding a loading state and setting it to `false` after `setUser` is called to ensure the loader is hidden once the user data is available.